### PR TITLE
Make sure ResetTokenBackgroundJob doesn't execute if config is read-only

### LINKED ISSUE #32290: [Bug]: config_is_read_only: Background job wants to modify config.php
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In the nextcloud.log the background job wants to modify the config.php which is set to read only. However it wants to delete a parameter from it. The error message doesn't tell you which parameter it wants to delete so I can manually edit it.

See log entry below.

### Steps to reproduce

1. Set `config_is_read_only` to `true` in config.php
2. Wait for the background job to run
3. Check nextcloud.log

### Expected behavior

The error message should be more useful an tell which config option to modify or delete!

### Installation method

Other

### Operating system

Other

### PHP engine version

Other

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

_No response_

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{                                                                                                                                                                           [10/265]
    "system": {                                                                           
        "installed": true,                                                                
        "config_is_read_only": true,
        "dbtype": "pgsql",                                                                
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***", 
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "cld_",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "theme": "",                         
        "defaultapp": "files",
        "knowledgebaseenabled": true,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": false,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "appstoreenabled": false,
        "appcodechecker": true,
        "apps_paths": [
            {
                "path": "\/srv\/www\/vhosts\/nextcloud\/public\/apps",
                "url": "\/apps",
                "writable": false
            }
        ],
        "updatechecker": true,
        "updater.release.channel": "stable",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "version": "24.0.0.12",
        "maintenance": false,
        "trusted_domains": [
            "cloud.cryptomilk.org"
        ],
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "logfilemode": 432,
        "loglevel": 2,
        "overwrite.cli.url": "https:\/\/cloud.cryptomilk.org",
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        }
    }
}
```

### List of activated Apps

```shell
- accessibility: 1.10.0                                                                                                                                                   [11/325]
  - admin_audit: 1.14.0
  - calendar: 3.2.2      
  - circles: 24.0.0      
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0    
  - dav: 1.22.0          
  - deck: 1.7.0 
  - dicomviewer: 1.2.3
  - federatedfilesharing: 1.14.0             
  - federation: 1.14.0
  - files: 1.19.0
  - files_markdown: 2.3.6
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - groupfolders: 12.0.0
  - impersonate: 1.11.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - notify_push: 0.4.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - richdocuments: 6.0.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - workflowengine: 2.6.0
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

```shell
{"reqId":"trtJ8hLDMaulApeUtrl0","level":3,"time":"2022-05-06T11:18:52+00:00","remoteAddr":"","user":"--","app":"core","method":"","url":"--","message":"Error while running background job (class: OCA\\UpdateNotification\\ResetTokenBackgroundJob, arguments: )","userAgent":"--","version":"24.0.0.12","exception":{"Exception":"OCP\\HintException","Message":"Config is set to be read-only via option \"config_is_read_only\".","Code":0,"Trace":[{"file":"/srv/www/vhosts/nextcloud/public/lib/private/Config.php","line":188,"function":"checkReadOnly","class":"OC\\Config","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/lib/private/Config.php","line":174,"function":"delete","class":"OC\\Config","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/lib/private/SystemConfig.php","line":189,"function":"deleteKey","class":"OC\\Config","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/lib/private/AllConfig.php","line":188,"function":"deleteValue","class":"OC\\SystemConfig","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/apps/updatenotification/lib/ResetTokenBackgroundJob.php","line":63,"function":"deleteSystemValue","class":"OC\\AllConfig","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/lib/private/BackgroundJob/Job.php","line":54,"function":"run","class":"OCA\\UpdateNotification\\ResetTokenBackgroundJob","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/lib/private/BackgroundJob/TimedJob.php","line":60,"function":"execute","class":"OC\\BackgroundJob\\Job","type":"->"},{"file":"/srv/www/vhosts/nextcloud/public/cron.php","line":151,"function":"execute","class":"OC\\BackgroundJob\\TimedJob","type":"->"}],"File":"/srv/www/vhosts/nextcloud/public/lib/private/Config.php","Line":298,"Hint":"Unset \"config_is_read_only\" to allow changes to the config file.","CustomMessage":"Error while running background job (class: OCA\\UpdateNotification\\ResetTokenBackgroundJob, arguments: )"}}
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by tcitworld**
Closes #32290

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Review by kesselb — APPROVED** [review]

**Comment by tlemarchand** [comment]
Thank you @tcitworld .
Nextcloud team, can this PR be merged and backported to 24 ?
Read-only root FS is a very common use case in containers (and a best practice), because of this upgrade to 24 is difficult right now.

File: apps/updatenotification/lib/ResetTokenBackgroundJob.php [hunk 1]
```diff
@@ -59,7 +60,7 @@ public function __construct(IConfig $config,
 	 */
 	protected function run($argument) {
 		// Delete old tokens after 2 days
-		if ($this->timeFactory->getTime() - $this->config->getAppValue('core', 'updater.secret.created', $this->timeFactory->getTime()) >= 172800) {
+		if ($this->config->getSystemValueBool('config_is_read_only') === false && $this->timeFactory->getTime() - $this->config->getAppValue('core', 'updater.secret.created', $this->timeFactory->getTime()) >= 172800) {
```
Comment (CarlSchwan) [inline apps/updatenotification/lib/ResetTokenBackgroundJob.php:63]:
```suggestion
		if ($this->config->getSystemValueBool('config_is_read_only') === false && $this->timeFactory->getTime() - (int)$this->config->getAppValue('core', 'updater.secret.created', (string)$this->timeFactory->getTime()) >= 172800) {
```

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**